graphitron-codegen-parent/graphitron-java-codegen/src/main/java/no/sikt/graphitron/generators/codebuilding/LookupHelpers.java

```diff
@@ -165,10 +166,19 @@ private static CodeBlock buildKey(String key, ObjectField ref, ProcessedSchema s
             return GeneratorConfig.shouldMakeNodeStrategy()
                     ? createNodeIdBlock(schema.getObject(it.getNodeIdTypeName()), table.toString())
                     : CodeBlock.of("$L$L", table, it.getMappingFromFieldOverride().asGetCall());
+        } else if (!it.getTypeClass().equals(ClassName.get(String.class))) {
+            return CodeBlock.of("$L.$L.cast($T.class)", table, it.getUpperCaseName(), STRING.className);
         }
         return CodeBlock.of("$L.$L", table, it.getUpperCaseName());
```

Comment:
Alternativt noe i denne retningen:
```suggestion
        return CodeBlock.of(
                 "$L.$L$L",
                 table,
                 it.getUpperCaseName(),
                 !it.getTypeClass().equals(ClassName.get(String.class)) ? CodeBlock.of(".cast($T.class)", STRING.className) : CodeBlock.empty()
        );
```
eller
```suggestion
        var fieldBlock =  CodeBlock.of("$L.$L", table, it.getUpperCaseName());
        if (it.getTypeClass().equals(ClassName.get(String.class))) {
            return fieldBlock;
        }

        return CodeBlock.of("$L.cast($T.class)", fieldBlock, STRING.className);
```
Trenger i hvert fall ikke else siden det er return i alle.